### PR TITLE
Assignment

### PR DESCRIPTION
**Pull request by cthulahoops**
First steps in this. Need to generalise to support other iterators not just tuples.

### CODE REVIEW

File: parser/src/parser.rs [hunk 1]
```diff
@@ -253,6 +253,40 @@ mod tests {
...
+    #[test]
+    fn test_parse_tuples() {
+        let source = String::from("a, b = 4, 5\n");
+
+        assert_eq!(
+            parse_statement(&source),
+            Ok(ast::LocatedStatement {
+                location: ast::Location::new(1, 1),
+                node: ast::Statement::Assign {
+                    targets: vec![ast::Expression::Tuple {
```
Comment (windelbouwman) [inline parser/src/parser.rs]: What kind of expressions are allowed at the left hand side (LHS)? Is it tuples, names only? Or are lists also allowed?

File: vm/src/vm.rs [hunk 2]
```diff
@@ -1045,6 +1046,19 @@ impl VirtualMachine {
                 }
                 None
             }
+            bytecode::Instruction::UnpackSequence { size } => {
+                let value = self.pop_value();
+
```
Comment (windelbouwman) [inline vm/src/vm.rs]: Should we add a type check here?

**Comment by windelbouwman** [comment]
Nice job!

File: vm/src/vm.rs (hunk 2 above)
Comment (cthulahoops) [reply vm/src/vm.rs]: The full implementation requires ```__iter__```, which you've just started to provide. I'll extend this to call that.

File: parser/src/parser.rs (hunk 1 above)
Comment (cthulahoops) [reply parser/src/parser.rs]: Tuples, lists, names, attributes, indices, slices.

I'm not sure if there are any others.

**Pull request merged**